contracts/src/upgrade/GatewayWithFeeInitializer.sol

```diff
@@ -0,0 +1,37 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2023 Snowfork <hello@snowfork.com>
+pragma solidity 0.8.28;
+
+import "../Gateway.sol";
+
+// New Gateway logic contract with an fee initializer
+contract GatewayWithFeeInitializer is Gateway {
```

Comment:
I'm trying for us to have a naming style like `Gateway202508`, based on the month in which we intend the upgrade to to be deployed.

```suggestion
contract Gateway202508 is Gateway {
```